### ISSUE TITLE
Extended variable assignment doesn't work if object has a list.

### ISSUE THREAD

**Issue by VitaliiBilash** [robotframework/robotframework#3965]
I need to set variables using extended variable syntax. I have a case where object has a list. Until release 4rc1 I could do that, but now I can't.
Here example.

```
Input new value
    ${object_to_add}=       Create Dictionary   name=old value
    ${array_to_add}=        Create List         ${object_to_add}
    &{body}=                Create Dictionary   data=${array_to_add}
    log to console    \nold body ${body}
    ${body.data[0].name}    set variable        new value
    log to console    new body ${body}
```
In that example, I receive 

`Variable '${body}' does not have attribute 'data[0]'`

In RF 4b3 (or earlier):
```
old body {'data': [{'name': 'old value'}]}
new body {'data': [{'name': 'new value'}]}
```

**Comment by pekkaklarck** [comment]
Thanks for the report. I was able to reproduce it and based on `git bisect` the commit that broke this is 0cd538d9149fbff2292911fa308c9bffddac8072. I try to get this fixed for RF 4.0.3.

**Comment by pekkaklarck** [comment]
The problem was due to 0cd538d9149fbff2292911fa308c9bffddac8072 changing extended assignment algorithm so that name like `${a.b.c}` was split from `.` and each attribute (`b` and `c`) got from the base variable individually using `getattr`. This worked fine in with "normal" attributes, but failed when using something more complicated like `${a.b[0].c}`.

Fixing the problem wasn't overly complicated in the end, but the fix removed detecting error in cases like

    ${var.nonex.attr} =    Set Variable    value

where `${var}` exists but doesn't have attribute `nonex`. The previous algorithm reported an error that `${var}` doesn't contains `nonex`, but now we just set a new variable with name `${var.nonex.attr}`. This is how the code worked before RF 4.0 and this is also consistent with now we handle

    ${nonex.attr} =    Set Variable    value

where `${nonex}` doesn't exist. I guess we can consider the current (and old) behavior fine. Some time in the future we probably should rewrite extended assign altogether and handle these cases differently. Then we could also support variants like `${var[0]}` that currently don't work at all.
